### ISSUE TITLE
CI should test the PR merged into the lastest develop instead of stale PR

### ISSUE THREAD

**Issue by panyx0718** [PaddlePaddle/Paddle#10669]
We often see situations that CI is broken by a merged PR. However, unless the test is flaky, a merged PR shouldn't break the CI. This is because PR can only be merged after CI has passed.

We suspect that this is because the CI is testing the PR "as it is". The PR can be 10 days stale, and doesn't work well with the latest develop code. In other words, the PR can be found to be harmful only after it has been merged with the develop.

I'd like to propose a mechanism to mitigate the issue:
1. CI clones the latest code from develop branch.
2. CI merges the PR into the develop branch.
3. CI runs the tests based on the merged PR.
4. After CI has passed, the PR can be really merged.

**Comment by shanyi15** [comment]
您好，此issue在近一个月内暂无更新，我们将于今天内关闭。若在关闭后您仍需跟进提问，可重新开启此问题，我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解~感谢您对PaddlePaddle的支持! 
 Hello, this issue has not been updated in the past month. We will close it today for the sake of other user‘s experience. If you still need to follow up on this question after closing, please feel free to reopen it. In that case, we will get back to you within 24 hours. We apologize for the inconvenience caused by the closure and thank you so much for your support of PaddlePaddle Group!
